src/main/kotlin/org/rust/ide/formatter/RustFormattingModelBuilder.kt

```diff
@@ -3,19 +3,28 @@ package org.rust.ide.formatter
 import com.intellij.formatting.FormattingModel
 import com.intellij.formatting.FormattingModelBuilder
 import com.intellij.formatting.FormattingModelProvider
+import com.intellij.formatting.SpacingBuilder
 import com.intellij.lang.ASTNode
 import com.intellij.openapi.util.TextRange
 import com.intellij.psi.PsiElement
 import com.intellij.psi.PsiFile
 import com.intellij.psi.codeStyle.CodeStyleSettings
+import org.rust.lang.RustLanguage
+import org.rust.lang.core.psi.RustTokenElementTypes.COMMA
 
 class RustFormattingModelBuilder : FormattingModelBuilder {
     override fun getRangeAffectingIndent(file: PsiFile?, offset: Int, elementAtOffset: ASTNode?): TextRange? = null
 
     override fun createModel(element: PsiElement, settings: CodeStyleSettings): FormattingModel {
-        val block = RustFormattingBlock(element.node)
+        val block = RustBlock(element.node, settings, createSpacingBuilder(settings))
         return FormattingModelProvider.createFormattingModelForPsiFile(element.containingFile, block, settings)
     }
 
+    protected fun createSpacingBuilder(settings: CodeStyleSettings): SpacingBuilder {
+        val rustSettings = settings.getCustomSettings(RustCodeStyleSettings::class.java)
+        val commonSettings = settings.getCommonSettings(RustLanguage)
+        return SpacingBuilder(commonSettings)
+            .after(COMMA).spaceIf(commonSettings.SPACE_AFTER_COMMA)
+            .before(COMMA).spaceIf(commonSettings.SPACE_BEFORE_COMMA)
```

Comment:
Actually Go language plugin hardcodes all spacing options :wink: I'd prefer to make this as much configurable as possible, having in mind that making this things configurable is pretty easy.